x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -58,6 +58,11 @@ export const postActionsConnectorExecuteRoute = (
         // TODO: Add `traceId` to actions request when calling via langchain
         logger.debug('Executing via langchain, assistantLangChain: true');
 
+        // Fetch any tools registered by the request's originating plugin
+        const assistantTools = (await context.elasticAssistant).getRegisteredTools(
+          'securitySolution'
```

Comment:
As noted in our offline discussion, ideally this route in `x-pack/plugins/elastic_assistant/server/routes/evaluate/post_evaluate.ts` wouldn't need to reference (by name) the `securitySolution` plugin here.  (It's passing `securitySolution` to `getRegisteredTools` to ensure only tools registered by the security solution plugin are returned for this route.)

Not required for this PR, but if it's possible to extract the calling plugin from the http request context (via, for example Kibana-specific http request headers), it may be possible to remove the hard-coded reference to the security solution here. (And in similar routes.)

Reply:
I had asked the core folks over here ([slack](https://elastic.slack.com/archives/C5TQ33ND8/p1701332103798599?thread_ts=1701303436.257429&cid=C5TQ33ND8)) and they said it's not a 100% fit solution since there are corner cases with background plugins and 'helpers' making requests, but this should suffice given our use case. 


I've added a new function called `getPluginNameFromRequest`, and corresponding tests to resolve this issue. If an error is thrown while trying to deduce the plugin name, we'll fallback to `securitySolutionUI` which is the plugin name registered with the security routes. Perhaps in the future we just have a 'generic bin' of tools that can be the default.